### ISSUE TITLE
Must always supply credentials when adding repo despite common secret being present in repository.credentials

### ISSUE THREAD

**Issue by moensch** [argoproj/argo-cd#2305]
**Describe the bug**

I added credentials for a common repo prefix to `ConfigMap/argocd-cm:repository.credentials` under the URL `git@github.com:myorg/`. If I manually add a new repo to the `repositories` list and *just* provide the repo URL like `git@github.com:myorg/myrepo.git` then it works and Argo CD can deploy from that repo.

I would like to be able to add new repos to that list using the CLI or API but this always fails with:
`rpc error: code = Unknown desc = error creating SSH agent: "SSH agent requested but SSH_AUTH_SOCK not-specified"`

I am assuming that the server side validation code doesn't take "shared" credentials into account.

**To Reproduce**

If we cannot reproduce, we cannot fix! Steps to reproduce the behavior:

1. Add a new entry to the `argocd-cm` configmap like so referencing a private key secret that has access to all repositories within the specified GitHub organization:

```
  repository.credentials: |
    - sshPrivateKeySecret:
        key: sshPrivateKey
        name: somesecret
      url: git@github.com:myorg/
```
2. Call this via the CLI now:
`argocd repo add git@github.com:myorg/myrepo.git`

Or do this via the Go client:
```go
    repoAccessReq := repositorypkg.RepoAccessQuery{
        Repo: "git@github.com:myorg/myrepo.git",
    }
    newconn, repoIf := argocdclient.NewClientOrDie(clientOpts).NewRepoClientOrDie()
    defer util.Close(newconn)
    _, err = repoIf.ValidateAccess(context.Background(), &repoAccessReq)
    if err != nil {
        log.Printf("ERROR: %s\n", err)
    }
```

4. Both cases return
```
rpc error: code = Unknown desc = error creating SSH agent: "SSH agent requested but SSH_AUTH_SOCK not-specified"
```

**Expected behavior**

Should be able to add a new repo without supplying new credentials if existing credentials exist in `repository.credentials` in `argocd-cm`

**Screenshots**

**Version**

`v1.2.0`

```shell
argocd: v1.2.0+674978c
  BuildDate: 2019-09-04T21:27:44Z
  GitCommit: 674978cd587701b39e81fce6d5c960b6d76d5882
  GitTreeState: clean
  GoVersion: go1.12.6
  Compiler: gc
  Platform: darwin/amd64
argocd-server: v1.2.0+674978c
  BuildDate: 2019-09-04T21:27:17Z
  GitCommit: 674978cd587701b39e81fce6d5c960b6d76d5882
  GitTreeState: clean
  GoVersion: go1.12.6
  Compiler: gc
  Platform: linux/amd64
  Ksonnet Version: 0.13.1
```

**Logs**

```
Paste any relevant application logs here.
```

**Have you thought about contributing a fix yourself?**

Open Source software thrives with your contribution. It not only gives skills you might not be able to get in your day job, it also looks amazing on your resume. 

If you want to get involved, check out the 
[contributing guide](https://github.com/argoproj/argo-cd/blob/master/docs/CONTRIBUTING.md), then reach out to us on [Slack](https://argoproj.github.io/community/join-slack) so we can see how to get you started.

**Comment by alexec** [comment]
Fixed by #2305

**Comment by alexec** [comment]
Fixed by #2308.

**Comment by moensch** [comment]
Wow that was fast. I wasn’t sure how to proceed once I found #2207. Thank you!
